webknossos-datastore/app/com/scalableminds/webknossos/datastore/dataformats/BucketProvider.scala

```diff
@@ -11,30 +12,28 @@ import scala.concurrent.ExecutionContext
 
 trait BucketProvider extends FoxImplicits with LazyLogging {
 
-  def loadFromUnderlying(readInstruction: DataReadInstruction): Box[Cube] = Empty
+  // To be defined in subclass.
+  def loadFromUnderlying(readInstruction: DataReadInstruction): Box[WKWCube] = Empty
 
   def load(readInstruction: DataReadInstruction, cache: DataCubeCache)(
-      implicit ec: ExecutionContext): Fox[Array[Byte]] = {
-
-    def loadFromUnderlyingWithTimeout(readInstruction: DataReadInstruction): Box[Cube] = {
-      val t = System.currentTimeMillis
-      val result = loadFromUnderlying(readInstruction)
-      val duration = System.currentTimeMillis - t
-      if (duration > 500) {
-        val className = this.getClass.getName.split("\\.").last
-        logger.warn(
-          s"loading file in $className took ${if (duration > 3000) "really " else ""}long.\n"
-            + s"  duration: $duration\n"
-            + s"  dataSource: ${readInstruction.dataSource.id.name}\n"
-            + s"  dataLayer: ${readInstruction.dataLayer.name}\n"
-            + s"  cube: ${readInstruction.cube}"
-        )
-      }
-      result
+      implicit ec: ExecutionContext): Fox[Array[Byte]] =
```

Comment:
Why is it a `Fox[..]`? I think the DataCubeCache can be rewrote to only use `Box[..]`, but maybe that's another issue.

Reply:
Yes could be right, but since this is not a change introduced here, I’d defer that. The Fox does not do any harm here for now :)